systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java

```diff
@@ -689,6 +689,36 @@ void testScaleConnectS2IWithConnectorToZero() {
         assertThat(connectorStatus.getConditions().stream().anyMatch(condition -> condition.getMessage().contains("has 0 replicas")), is(true));
     }
 
+    @Test
+    void testScaleConnectS2ISubresource() {
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3).done();
+
+        KafkaConnectS2IResource.kafkaConnectS2I(CLUSTER_NAME, CLUSTER_NAME, 1)
+            .editMetadata()
+                .addToLabels("type", "kafka-connect-s2i")
+            .endMetadata()
+            .done();
+
+        int scaleTo = 4;
+        long connectS2IObsGen = KafkaConnectS2IResource.kafkaConnectS2IClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getObservedGeneration();
+        String connectS2IGenName = kubeClient().listPods("type", "kafka-connect-s2i").get(0).getMetadata().getGenerateName();
+
+        LOGGER.info("-------> Scaling KafkaConnectS2I subresource <-------");
+        LOGGER.info("Scaling subresource replicas to {}", scaleTo);
+        cmdKubeClient().scaleByName(KafkaConnectS2I.RESOURCE_KIND, CLUSTER_NAME, scaleTo);
+        DeploymentConfigUtils.waitForDeploymentConfigAndPodsReady(KafkaConnectS2IResources.deploymentName(CLUSTER_NAME), scaleTo);
+
+        LOGGER.info("Check if replicas is set to {}, naming prefix should be same and observed generation higher", scaleTo);
+        List<String> connectS2IPods = kubeClient().listPodNames("type", "kafka-connect-s2i");
+        assertThat(connectS2IPods.size(), is(4));
+        assertThat(KafkaConnectS2IResource.kafkaConnectS2IClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getReplicas(), is(4));
+        assertThat(KafkaConnectS2IResource.kafkaConnectS2IClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getReplicas(), is(4));
+        assertThat(connectS2IObsGen < KafkaConnectS2IResource.kafkaConnectS2IClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getObservedGeneration(), is(true));
+        for (String pod : connectS2IPods) {
```

Comment:
is this rly needed? I assume the s2i pods will contain it for sure. I see you have it everywhere.  I don't say it's wrong, maybe I just don't see the value of it?

Reply:
The main thing I add this to all tests is, that if you scale the replicas like earlier, you'll get completely different names, something like `my-cluster-connect-45456a-asd456, my-cluster-connect-weq56a-asd456`, but when you scale the sub-resources, the other replicas will have **same generated name** (this can be seen in the yaml of resource in field `generateName: ...` so the result be like `my-cluster-connect-45456a-asd456, my-cluster-connect-45456a-asd456` -> only the end will change. So I'm testing, that all replicas are really containing this `generateName` part.